packages/contracts/contracts/UnipigTransitionEvaluator.sol

```diff
@@ -0,0 +1,328 @@
+        );
+        // Now simply apply the transfer transition as usual
+        return applyTransferTransition(transferTransition, _storageSlots);
+    }
+
+    /**
+     * Apply a transfer stored account transition
+     */
+    function applyTransferTransition(
+        dt.TransferTransition memory _transition,
+        dt.StorageSlot[2] memory _storageSlots
+    ) public view returns(dt.Storage[2] memory) {
+        // First construct the transaction from the storage slots
+        address sender = _storageSlots[0].value.pubkey;
+        address recipient = _storageSlots[1].value.pubkey;
+        dt.TransferTx memory transferTx = dt.TransferTx(sender, recipient, _transition.tokenType, _transition.amount);
+
+        // Next check to see if the signature is valid
+        require(verifyEcdsaSignature(_transition.signature, getTransferTxHash(transferTx), sender), "Transfer signature is invalid!");
+        // Also make sure we're not sending to Unipig
+        require(_storageSlots[1].slotIndex != UNISWAP_SLOT_INDEX, "Transfer cannot be made to Unipig!");
+
+        // Create an array to store our output storage slots
+        dt.Storage[2] memory outputStorage;
+        // Now we know the pubkeys are correct, let's compute the output of the transaction
+        uint senderBalance = _storageSlots[0].value.balances[transferTx.tokenType];
+
+        // First let's make sure the sender has enough money
+        require(senderBalance > transferTx.amount, "Sender does not have enough money!");
+
+        // Update the storage slots with the new balances
+        _storageSlots[0].value.balances[transferTx.tokenType] -= transferTx.amount;
+        _storageSlots[1].value.balances[transferTx.tokenType] += transferTx.amount;
+        // Set the outputs
+        outputStorage[0] = _storageSlots[0].value;
+        outputStorage[1] = _storageSlots[1].value;
+        // Return the outputs!
+        return outputStorage;
+    }
+
+    /**
+     * Apply a swap transition
+     */
+    function applySwapTransition(
+        dt.SwapTransition memory _transition,
+        dt.StorageSlot[2] memory _storageSlots
+    ) public view returns(dt.Storage[2] memory) {
+        address sender = _storageSlots[0].value.pubkey;
+        address recipient = _storageSlots[1].value.pubkey;
+        // Create our swapTx
+        dt.SwapTx memory swapTx = dt.SwapTx(
+            sender,
+            _transition.tokenType,
+            _transition.inputAmount,
+            _transition.minOutputAmount,
+            _transition.timeout
+        );
+
+        // Make sure that the provided storage slots are corrent
+        require(verifyEcdsaSignature(_transition.signature, getSwapTxHash(swapTx), sender), "Swap signature is invalid!");
+        require(_storageSlots[1].slotIndex == UNISWAP_SLOT_INDEX && recipient == UNISWAP_ADDRESS, "Swap tx must be swapping with Unipig!");
+
+        // Create an array to store our output storage slots
+        dt.Storage[2] memory outputStorage;
```

Comment:
Would create this closer to where it's assigned, if it needs to be created at all.